incubator/hnc/internal/reconcilers/object.go

```diff
@@ -787,14 +793,37 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getTreeSelector returns the selector on a given object if it exists
+func (r *ObjectReconciler) getTreeSelector(log logr.Logger, inst *unstructured.Unstructured) labels.Selector {
+	annot := inst.GetAnnotations()
+	treeSelectorStr, ok := annot[api.AnnotationTreeSelector]
+	if !ok {
+		return nil
+	}
+	strs := strings.Split(treeSelectorStr, ",")
```

Comment:
Comment why we're doing this. It's fine to refer to the design doc, but it might also make sense to document this here as well.

The comment might make more sense at the function level (e.g. at line 796). Eg "A tree selectors is similar to a regular selector, except for the following things... To transform a tree selector into a regular label selector, we follow these steps..."

Reply:
done